### DISCUSSION TITLE
Inset to current openned note

### DISCUSSION THREAD

**Discussion by mpietrusa** [coddingtonbear/obsidian-local-rest-api#34] [Ideas]
Hi,
I use web clipper and it would be nice if I could invoke rest-api-local to add contents to current openned note.
If I correctly analyzed functions for local-rest-api then this is not supported in current version. Please add it.

**Comment by coddingtonbear** [comment]
You're right that it wasn't supported.  _Usually_, I'd ask you, @mpietrusa, to put together a PR supporting your use case, but this particular one could sometimes be useful for me, too, and I looked into this briefly and it turned out to be very easy.  So: I did it for you.  This feature exists now as part of v1.5.0 of this plugin via the `/active/` REST endpoint.  See the docs for more details.

**Reply by mpietrusa** [reply]
Hi,
Many thanks for your changes. 
I have tested and it works great.

**Comment by ebw44** [comment]
Is there a way for the insertion by PATCH to append at the current position of the cursor instead of a specific header?

**Reply by mjund** [reply]
i think this would be helpful! is it possible? we would be happy to do the work if @coddingtonbear guide us on how to do it!

**Reply by ebw44** [reply]
I add a quick look a the code with the chrome debugger and it looks like it is not straightforward as it seems to get the text from the currently open file, then does the append or inserts, and finally update the file content. So it has no access to the cursor.  

My current workaround is to insert a template heading manually (in my case `###### Insert`) and use the current PATCH method.  

I tried to look at using a command but I couldn't find a way to pass an argument to a command. If there is a way, it may be a solution to implement the feature. Even if using the cursor position in the regular PATH /active/ call (for example when heading is missing) seems like a more logical way for me as a user.

**Reply by coddingtonbear** [reply]
FWIW, I do not have time to think about this right now in any depth, but I'm generally convinced that this would be a useful feature.  Once I have a little more time on my hands, I imagine I might re-work the `PATCH` handler to allow other kinds of references in addition to header references (see also #26); so if we all just hold on for a little while, I think I'll eventually be able to get to this.
